src/lib.rs

```diff
@@ -132,31 +131,34 @@ where
     #[cfg(feature = "table-decoder")]
     /// Call `update` to evaluate the next state of the encoder, propagates errors from `InputPin` read
     pub fn update(&mut self) -> Result<Direction, Either<A::Error, B::Error>> {
-        #[cfg(not(feature = "embedded-hal-alpha"))]
-        let (a_is_high, b_is_high) = (self.pin_a.is_high(), self.pin_b.is_high());
-        #[cfg(feature = "embedded-hal-alpha")]
         let (a_is_high, b_is_high) = (self.pin_a.is_high(), self.pin_b.is_high());
 
         // Implemented after https://www.best-microcontroller-projects.com/rotary-encoder.html
         self.prev_next <<= 2;
         if a_is_high.map_err(Either::Left)? {
-            self.prev_next |= 0x02;
+            self.prev_next |= 0x01;
```

Comment:
I have a local change which I'll submit after this which benefits from using the approach in the article, so I swap these values and the values above, which preserves behavior.

Reply:
Maybe I don't understand what you're getting at here. by why swap the A/B pins then swap the direction? The `From` impl is a public method, and this change would still break anyone using it?

Reply:
I have another PR that removes the From impl and adds a customization point (trait) and it was easier to go with the numbers in the article, which assume A as LSB and B as MSB.

I have https://www.adafruit.com/product/5001 just for fun. The datasheet says that one full 00 -> 00 is 30 or 12 per complete rotation. However, physically, there are 24 detents, which means half a full 00 -> 00.

When using the regular implementation (without the table-decoder feature), I can create bounces, so having the table-decoder to avoid that is beneficial. Unfortunately, the current table-decoder only recognizes a move once per 00 -> 00 and on a specific one. This means that I can't recognize the half of the detents that the user produces. And for this encoder, the current implementation doesn't recognize the first detent when reversing.

So, the solution is that I create a `Phase` trait with a single function that replaces the From impl and lets the user decide which values should return None or not. That way, I can pick the two phases of the four that result in every detent being recognized on the encoder while also recognizing reversals between two detents.

I also have a PRs which add proptest support (though only for table-decoder, since I'd have to write a completely different oracle for the other mode) and reduce the state for table-decode to a single `u8`.

Anyways, I have some goals:
1. I'd prefer to improve an existing library over making my own.
2. Have support for debouncing.
3. Have the customization point so it works will with this encoder I have.

I don't think it's possible to do 3 without a breaking change of some form, unfortunately. But ultimately, it's up to you about which changes you want to merge and how to deal with possible breaking changes.

Reply:
Thanks for the explanation. I'd be open to merging a trait, provided the library remains easy to use for the average user. As far as these changes, I'll hold off on releasing them until From is removed

Reply:
Are you ok with this PR as it is or would you like any specific changes? Thanks.